### ISSUE TITLE
WiCAN stops mqtt sendings and needs re-plugging to send again

### ISSUE THREAD

**Issue by micha53** [meatpiHQ/wican-fw#274]
I got my WiCAN this week and it works in principle. I use it in a Renault Zoe Ph. 2.
But I have one Problem: If the car ist standing and not connected to a wallbox, it stops sending mqtt telegrams. For  a while it sends an Error Information, later this also stops. This should not be a problem, when it would start sending per mqtt after starting to load the car. But the stick remains "Sleeping" when I start loading the car. (It is the same behavior with and without enabling sleep mode in the settings.) I think in this moment the stick should restart to send mqtt.
A Reboot of the stick makes no effect, only disconnecting and replugging brings success. 
I do not understand why replugging (hard reset) works but a software reboot brings no effect. But in general the stick should restart to send mqtt, when the loading process starts. I can't imagine that replugging can start the CAN communication of the car, so why does the stick not listen to the CAN?
The ...can/status topic says "offline" and the ...can/rx is not present.
At least with enabled sleep mode the power led is blinking and I cannot access the stick until I replug it. Also turning on the ignition of the car brings no change.
I would like to have that the wican starts to send mqtt if I start to load the car, without replugging the stick.

**Comment by meatpiHQ** [comment]
can you please share a screenshot for your device configuration page? And the status tab as well.

**Comment by micha53** [comment]
Config and Status:
![Screenshot_20241206_164859_Firefox](https://github.com/user-attachments/assets/39c238c4-c4ec-4c1a-bf09-26bf17da1fce)
![Screenshot_20241206_165009_Firefox](https://github.com/user-attachments/assets/37c6c55f-b6d3-4bac-be00-bfde27c83ccc)
![Screenshot_20241206_165034_Firefox](https://github.com/user-attachments/assets/63001987-f839-4394-bcac-820f069efaa0)

**Comment by meatpiHQ** [comment]
It's normal to stop sending MQTT if the car is not connected to the charger, because the ecu goes in low power mode. But it sounds like there is a bug in the firmware, when it's connected but ecu is not responding it fails. 

I will try to reproduce this issue here.

What happens if, when in this state you plug in the charger and then reboot?

**Comment by micha53** [comment]
actually this happened:
i was coming home by car half an hour before. Just looked at the dongle: Blue and green LED' s were on, yellow one was flashing. I started the car and moved some meters. After that flashing the yellow LED stopped.
And now, additional 15 minutes later I opened the mqtt explorer and for some cycles I saw the right values and after some seconds they vanished  and I see only this:
![2024-12-06 17_38_35-Window](https://github.com/user-attachments/assets/eedaeea4-6aeb-4ce5-a626-7b5d42ffa8c4)
**cfos_mqtt/set/M12** is the automate destination topic.

**Your Question:
If I plug in the charger and reboot nothing changes in the mqtt explorer.**

btw. using the monitor in this situation I do not see any can-message.

**Comment by micha53** [comment]
I made another test today:
Wican is in sleep mode and sleeps, I open the door: Wican wakes up and publishes car data
I close the door: after one minute wican stops publishing and goes to sleep
I open the door again and open the charging connector and close the door again and plugin the charging cable: Wican wakes up and publishes car data

So the adapter works in a way like it should do, only I had done something that caused the error. But actually I cannot say what was the difference.

I will continue to observe the behavior and will give a feedback when I know more about it. So actually there's nothing to do for the developer.

**Comment by micha53** [comment]
Actually it works fine, so I close this issue.

**Comment by meatpiHQ** [comment]
@micha53 works fine on the latest firmware version or are you still on the old one?

**Comment by micha53** [comment]
Ah, I see there is a new version from yesterday.. No I still have the old one.
But a question concerning the new one: 
I had problems with the value for the 12VBatt Voltage. In home assistant this variable causes an error. Adding a letter bevore the 12 solved the problem, but does not read the value published by the wican. Maybe it is not good to begin a value used here with a number. Does it still have the name 12vBatt?

**Comment by meatpiHQ** [comment]
That's in the zoe vehicle profile right? I can change it in the profile, this is not hard coded in the firmware.

**Comment by micha53** [comment]
yes in the zoe profile

**Comment by meatpiHQ** [comment]
Ok, it's fixed now. Resync the profile and give it a go.

**Comment by micha53** [comment]
thank you
and.. can you tell me, or put into the doc, what is the payload of an Battery alert?

**Comment by meatpiHQ** [comment]
here you go

{"alert": "low_battery", "battery_voltage": 12.0}

**Comment by micha53** [comment]
so, Christmas is over, I thank you for the presents and I wish you all the best and a happy new year!

**Comment by meatpiHQ** [comment]
Happy new year! Cheers!

**Comment by micha53** [comment]
Last december I closed this Issue. 
But from time to time it happens that the behavior I described here comes back. I can't recognize any pattern when or why it occures. 
Meanwhile I made an update to Ver. 4.12 and it still happens from time to time that the device
- wakes up and the blue led comes
- connects to the AP of my home network and the green LED comes
- but it does not send mqtt messages/the yellow LED flashing does not start flashing 
When I
- was driving the car between,
- open the car door to unlock the flap to the charge connector
- and connect the chargers cable to the car and
- start charging
so I do not get an Update about the SOC of the cars traction battery.

When necessary I can repeat the test with rebooting in this state.

**Comment by micha53** [comment]
I had today the situation again, made a reboot, but the wican does not send mqtt messages after the reboot. Also a 2nd reboot did not change the behavior.

**Comment by meatpiHQ** [comment]
@micha53 Did you try to enable `Auto AP Disable`?

**Comment by micha53** [comment]
Yes, I enabled this immediately with th update to 4.12. It is a very smart feature.

**Comment by meatpiHQ** [comment]
@micha53 please try this firmware version. I suspect the ECU polling is failing, and not the MQTT.

[wican-fw_obd_v412-63-gc8d257c.zip](https://github.com/user-attachments/files/20709232/wican-fw_obd_v412-63-gc8d257c.zip)

**Comment by micha53** [comment]
I made the update and after the update at first it does not send mqtt. But after a reboot mqtt is working.
I will watch the behaviour the next days.

**Comment by meatpiHQ** [comment]
@micha53 do you still see the issue after the update?

**Comment by micha53** [comment]
After the update several times I opened the door, every time the wican woke up, connected and started to send mqtt.
Also after driving away one time the wican had the same behaviour, was ok.
But yesterday, after driving the car away from home (the wican wakes up, connects to my wifi, and maybe also started sending but the last fact I can't verify (my wife was driving the car) and after she came back, stopped the car, closed the doors and the wican sometime later went to sleep. Today I wanted to charge the car, opened the door, connected the car to the wallbox, meanwhile the wican woke up, connected to my wifi, but did not send mqtt again.
This evening I had the issue the 2nd time, again, after driving away with the car.
So yes, I still see the issue after the update.
It seems, and iI'll try to verify it, that it happens, after we drive the car away from home and come back, so the following happens:
- we open the door
- wican wakes up and connects to my wifi, maybe also already starts sending mqtt
- driving away the wican loses the connection to the wifi
- returning home it connects to the wifi and maybe also starts mqtt
- sometime later it goes in sleep state

**Comment by micha53** [comment]
little update, actually verified

- returning home it connects to the wifi and  **starts mqtt**

**Comment by micha53** [comment]
Yesterday I took a look into the wican code and found that there exist log entries. If I can support you anyhow with log data then give me instructions how to get them.

**Comment by micha53** [comment]
I had disabled sleep mode since 2025-06-23. Today coming back from a holiday week and found the Zoe with an empty 12V-battery. During this week the Zoe was standing and not attached anyhow.
What is the power consumption of the wican device in normal (not sleeping) mode? 
I could measure around 1A flowing out of the 12V-battery in closed car (doors) when the wican was plugged, without the wican only some mA. But I can't imagin that the wican takes 1A from the 12V-Bat.
I assume that my setting with 3000ms is too strong and the so wican causes some parts of the car electronics to remain acitve and they all in sum need the energy. With 1A I understand that the battery with 50Ah is empty after 2 days.
So I would like to make a proposal for a new feature: Polling the values dependend on the value of another topic. For example, most value are not interesting while no cable is plugged or charging current flows and need not to be polled most time, only when some special values indicate that the car will be charged. 
The odometer value for me is only from interest when I start or end driving.
